Add docs for multi repo pipelines (#1142)
gruntwork-io/docs: _docs-sources/pipelines/security/multi-account.md

> # Multiple Infrastructure-Live Repos

Reviewer: Nit: I personally find it easier to review and maintain the content when it's all on one line versus us manually adding line breaks.

Author: sorry, bad habit from Garmin where they very strictly enforced the 80 character line limit even in text files

Reviewer: Oh interesting. Yeah, let's standardize on one line then.

> We recommend using a single `infrastructure-live` repository for managing your organization's infrastructure.

Reviewer: ```suggestion
We recommend using a single `infrastructure-live` git repository for managing your organization's infrastructure.
```

Sometimes, this isn't possible due to team structure, security requirements, or other limitations.

> You may choose to use multiple repos to:

Reviewer: ```suggestion
You may choose to use multiple `infrastructure-live` repos to:
```

1. Facilitate more granular access controls
1. Separate concerns that do not require shared configuration
1. Ease the burden of high traffic repos (reducing the likelihood of feature branches becoming out-of-date relative to `main`)

> However, when using multiple repos it is more difficult to share configuration across environments so think carefully about
> your specific use case before making the decision.
> In order to accommodate multiple infrastructure repositories, Gruntwork Pipelines is configurable.

Reviewer: ```suggestion
The most common business situation that triggers these needs is having multiple teams. Fundamentally, you will need to decide if all teams will share a single `infrastructure-live` repository, if each team should have their own repository, or somewhere in the middle.

Note that when using multiple repositories, it is more difficult to share a Pipelines configuration across environments, so think carefully about your specific use case before making the decision.

To add multiple `infrastructure-live` repositories to a single Gruntwork Pipelines configuration, continue reading.
```

## Create Additional Repos

New `infrastructure-live` repositories can be created using the same process described in the
[Hello World](../hello-world#setting-up-the-repositories) documentation.

:::info

> Once the repo is created, you'll need to set up machine user access using either the existing machine user and `PIPELINES_DISPATCH` PAT token,

Reviewer: Personally, I think we should use the word "repo" instead of "repository" throughout the docs, but either way we should be consistent, so I guess this should probably be "repository."

or one created specifically for this purpose. See [Machine Users](../using-pipelines/machine-users) for more information.
:::

No special configuration is required for the new `infrastructure-live` repository,
the Pipelines Dispatch job will identify the source repository and pass that information
to the shared `infrastructure-pipelines` repository.

## Enable Additional Repos

:::warning

> Once a repo is enabled for pipelines, any code pushed to the `main` branch of that repo will be eligible to access your
> AWS account using OIDC. Ensure you have the [recommended settings](../using-pipelines) for branch protection configured before adding the new
> repository to the allowlist.

Reviewer: ```suggestion
Once an `infrastructure-live` repo is enabled for Pipelines, any code pushed to the `main` branch of that repo will have access to your
AWS account using OIDC. Ensure you have the [recommended settings](../using-pipelines) for branch protection configured before adding the new
repository to the allowlist.
```

Reviewer: "recommended settings" link is broken.

Side note: Do we have some automatic way of checking broken links using docusaurus? It seems hard to maintain manually.

Author: oh yes the build fails, I just missed it because it doesn't fail when using hot reload

:::

To ensure no unauthorized access is granted to your `infrastructure-pipelines` repository,
an allowlist of `infrastructure-live` repositories exists in the `.gruntwork/config.yml` file in the `infrastructure-pipelines` repository.
To allow resources to be deployed by your new repository,
add the repository to the `repo-allow-list` section of `.gruntwork/config.yml`.

The new resource should match the name of your repository **exactly** in the format
`github-org/infrastructure-live-repo-name` with a single repository per line. See the example file below:

> ```txt title=infrastructure-pipelines/.gruntwork/config.yml

Reviewer 2: nit: can we use `yaml` or `yml` to get syntax highlighting?
